File: src/easy/find-missing-and-repeated-values.ts
export const findMissingAndRepeatedValues = (grid: number[][]): number[] => {
	const n = grid.length;

	let expected = n * n * (Math.pow(n, 2) + 1) / 2;
	let repeated = 0;
	
	const freq: Record<number, number> = {}
	for (const cell of grid.flat(Infinity) as number[]) {
		freq[cell] = (freq[cell] || 0) + 1
		if (freq[cell] > 1) {
			repeated = cell;		
		} else {
			expected -= cell;
		}
		
	}

	return [repeated, expected]
}

File: easy/find_missing_and_repeated_values.py
def find_missing_and_repeated_values(
    grid: list[list[int]],
) -> list[int]:
    store = set()
    n = len(grid)
    repeated = 0
    expected = (pow(n, 2) * (pow(n, 2) + 1)) // 2
    for row in grid:
        for cell in row:
            if cell in store:
                repeated = cell
            else:
                store.add(cell)
                expected -= cell

    return [repeated, expected]
